src/util/EntityToFormMapper.js

```diff
@@ -385,28 +386,39 @@ const generateFormData = (fields: any, data: any, attributes: any) => {
       return accumulator
     }
 
+    const setDefaultValues = options.mapperMode === 'CREATE'
+
     switch (field.type) {
       case 'field-group':
         // Recursively generate data for compounds
-        return {...accumulator, ...generateFormData(field.children, data, attribute.attributes)}
+        return {...accumulator, ...generateFormData(field.children, data, attribute.attributes, options)}
       case 'file':
         // Map MOLGENIS FileMeta entity to our form file object
         // which only contains a name
         const fileData = data[field.id]
-        accumulator[field.id] = fileData ? fileData.filename : data[field.id]
+        if (setDefaultValues) {
+          accumulator[field.id] = attribute.defaultValue && attribute.defaultValue.filename
```

Comment:
I'd expect defaultValue to be a string, how can it have a filename attribute?

Reply:
i expected it would contain a file, but if every thing is a string this is not the case, i will change it.